src/main/kotlin/org/jitsi/jicofo/xmpp/AuthenticationIqHandler.kt

```diff
@@ -0,0 +1,59 @@
+/*
+ * Jicofo, the Jitsi Conference Focus.
+ *
+ * Copyright @ 2015 Atlassian Pty Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.jicofo.xmpp
+
+import org.apache.commons.lang3.StringUtils
+import org.jitsi.jicofo.auth.AuthenticationAuthority
+import org.jitsi.utils.logging2.createLogger
+import org.jitsi.xmpp.extensions.jitsimeet.LoginUrlIq
+import org.jitsi.xmpp.extensions.jitsimeet.LogoutIq
+import org.jivesoftware.smack.packet.IQ
+import org.jivesoftware.smack.packet.XMPPError
+
+class AuthenticationIqHandler(val authAuthority: AuthenticationAuthority) {
+    private val logger = createLogger()
+
+    fun handleLoginUrlIq(loginUrlIq: LoginUrlIq): IQ {
+        val peerFullJid = loginUrlIq.from.asEntityFullJidIfPossible()
+        val roomName = loginUrlIq.room
```

Comment:
Looks like it might be worth writing some helpers to create error responses.  If we had them we could do something like:
```
val roomName = loginUrlIq.room ?: return createNotAcceptableErrorResponse(loginUrlIq)
```

with helpers like:

```
fun createErrorResponse(iq: IQ, condition: XMPPError.Condition): IQ = IQ.createErrorResponse(iq, XMPPError.getBuilder(condition))

fun createNotAcceptableErrorResponse(iq: IQ): IQ = createErrorResponse(iq, XMPPError.Condition.not_acceptable)
```